src/components/form-helpers/form-helpers.js

```diff
@@ -52,6 +52,46 @@ export const FieldItem = connect(
   ),
 );
 
+export const PasswordWrapper = connect(
+  ({ className, fieldName, children, formik: { errors, touched } }) => (
+    <li
+      className={concatClasses(className, touched[fieldName] && errors[fieldName] ? 'error' : '')}
+    >
+      {children}
+      <div className="wrapper-password">
+        <p className="password-message">
+          <span
+            className={
+              !touched[fieldName]
+                ? 'waiting-message'
+                : touched[fieldName] &&
+                  errors[fieldName] &&
+                  (errors[fieldName].charLength || errors[fieldName].empty)
+                ? 'lack-message'
+                : 'complete-message'
+            }
+          >
+            <FormattedMessage id="validation_messages.error_password_character_length" />
```

Comment:
_FormattedMessage_ already creates a _span_ element, maybe you could apply the class to this element.

Reply:
No can do. https://github.com/yahoo/react-intl/issues/704 I've tried but seems to be not allowed.

Reply:
Oh, sorry, you are RIGHT!

I was really convinced that it just works, see this code:

https://github.com/FromDoppler/doppler-webapp/blob/634eda7080d234d8a9041feba52efe4ba7822eed/src/components/Loading/Loading.js#L9

Well, I think that without flash it is better haha.

Reply:
The other option is using _FormattedMessage_ as a wrapper, or simply having two spans as we have now.